### PR TITLE
fix(ecmascript): Array.prototype is an Array

### PR DESCRIPTION
**Pull request by aapoalas**
Built on top of #301 so should merge that first.

This mostly just makes `Array.prototype` an empty array exotic object itself. Stupid thing :D

But I also removed an unnecessary Vec allocation from element array grow code path, and updated the Rust toolchain to 1.79.0.

### CODE REVIEW

**Comment by andreubotella** [comment]
**question:** Why are the changes to the heap needed? Are they relevant to `Array.prototype`, or could be split out into a separate PR?

**Comment by aapoalas** [comment]
> **question:** Why are the changes to the heap needed? Are they relevant to `Array.prototype`, or could be split out into a separate PR?

Could definitely be split. Just what I happened to run into while I was looking at this.

**Comment by aapoalas** [comment]
I split the PR into three separate ones, see #308 and #309

File: nova_vm/src/heap/element_array.rs
Comment (aapoalas) [inline nova_vm/src/heap/element_array.rs:1]: Hmm, this is refactoring that got left behind apparently

**Review by andreubotella — COMMENTED** [review]
The `Array.prototype` changes LGTM.

**Review by andreubotella — APPROVED** [review]

**Pull request merged**